qemu/tests/balloon_service.py

```diff
@@ -97,42 +100,50 @@ def memory_check(vm, get_polling_output, keyname):
                                       (keyname, guest_mem, stat_memory_qmp,
                                        check_mem_ratio))
 
-    def balloon_memory(session, device_path):
+    def balloon_memory(vm, device_path, mem_check):
         """
         Doing memory balloon in a loop and check memory status during balloon.
 
-        :param session: VM session.
+        :param vm: VM object.
         :param device_path: balloon polling path.
+        :param mem_check: only need to do memory check if param mem_check
+                          is 'True'
         """
         repeat_times = int(params.get("repeat_times", 5))
         logging.info("repeat times: %d" % repeat_times)
-        balloon_test = BallooningTestWin(test, params, env)
-        balloon_test = BallooningTestWin(test, params, env)
+
+        if params['os_type'] == 'windows':
+            balloon_test = BallooningTestWin(test, params, env)
+        else:
+            balloon_test = BallooningTestLinux(test, params, env)
         min_sz, max_sz = balloon_test.get_memory_boundary()
         while repeat_times:
             for tag in params.objects('test_tags'):
                 error_context.context("Running %s test" % tag, logging.info)
                 params_tag = params.object_params(tag)
                 balloon_type = params_tag['balloon_type']
                 if balloon_type == 'evict':
-                    expect_mem = int(random.uniform(min_sz, balloon_test.get_ballooned_memory()))
+                    expect_mem = int(random.uniform(min_sz,
+                                     balloon_test.get_ballooned_memory()))
                 else:
-                    expect_mem = int(random.uniform(balloon_test.get_ballooned_memory(), max_sz))
+                    expect_mem = int(random.uniform(
+                        balloon_test.get_ballooned_memory(), max_sz))
 
                 quit_after_test = balloon_test.run_ballooning_test(expect_mem,
                                                                    tag)
                 time.sleep(20)
                 get_polling_output = vm.monitor.qom_get(device_path,
                                                         get_balloon_property)
                 time.sleep(20)
-                memory_check(vm, get_polling_output, 'stat-free-memory')
+                if mem_check is True:
+                    memory_check(vm, get_polling_output, 'stat-free-memory')
                 if quit_after_test:
                     return
 
             repeat_times -= 1
 
     timeout = int(params.get("login_timeout", 360))
-    driver_name = params.get("driver_name", "balloon")
+    mem_check = params.get("mem_check", True)
```

Comment:
Need convert to bool type.
"if mem_check is True: " is judging the bool type.

I think "yes" / "no" is better.